src/Cabinet/Cabinet.cs

```diff
@@ -0,0 +1,224 @@
+        /// </summary>
+        /// <param name="hf">
+        /// An application-defined value used to identify the open file.
+        /// </param>
+        /// <param name="pv">
+        /// Storage location for data.
+        /// </param>
+        /// <param name="cb">
+        /// Maximum number of bytes to read.
+        /// </param>
+        /// <returns>
+        /// The number of bytes read.
+        /// </returns>
+        [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
+        public unsafe delegate int FNREAD(int hf, void* pv, int cb);
+
+        /// <summary>
+        /// The <see cref="FNWRITE"/> macro provides the declaration for the application-defined callback function to write data to a file in an FDI context.
+        /// </summary>
+        /// <param name="hf">
+        /// An application-defined value used to identify the open file.
+        /// </param>
+        /// <param name="pv">
+        /// Data to be written.
+        /// </param>
+        /// <param name="cb">
+        /// Number of bytes.
+        /// </param>
+        /// <returns>
+        /// If successful, _write returns the number of bytes written.
+        /// </returns>
+        [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
+        public unsafe delegate int FNWRITE(int hf, void* pv, int cb);
+
+        /// <summary>
+        /// The <see cref="FNCLOSE"/> macro provides the declaration for the application-defined callback function to close a file in an FDI context.
+        /// </summary>
+        /// <param name="hf">
+        /// An application-defined value used to identify the open file.
+        /// </param>
+        /// <returns>
+        /// 0 if the file was successfully closed. A return value of -1 indicates an error.
+        /// </returns>
+        [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
+        public delegate uint FNCLOSE(int hf);
+
+        /// <summary>
+        /// The <see cref="FNSEEK"/> macro provides the declaration for the application-defined callback function to move a file pointer to the specified location in an FDI context.
+        /// </summary>
+        /// <param name="hf">
+        /// An application-defined value used to identify the open file.
+        /// </param>
+        /// <param name="dist">
+        /// Number of bytes from origin.
+        /// </param>
+        /// <param name="seektype">
+        /// Initial position.
+        /// </param>
+        /// <returns>
+        /// The offset, in bytes, of the new position from the beginning of the file.
+        /// </returns>
+        [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
+        public delegate uint FNSEEK(int hf, int dist, SeekOrigin seektype);
+
+        [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
+        public delegate int PFNNOTIFY(NOTIFICATIONTYPE fdint, NOTIFICATION fdin);
```

Comment:
You've done a great job with docs everywhere else. Is this one missing docs because no docs exist, or was it an oversight? This wouldn't be a reason to deny merging a PR, but I wanted to give you a chance to fill it in.